stdlib/public/core/Result.swift

```diff
@@ -0,0 +1,131 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// A value that represents either a success or failure, capturing associated
+/// values in both cases.
+@_frozen
+public enum Result<Success, Failure: Error> {
+  /// A success, storing a `Success` value.
+  case success(Success)
+  
+  /// A failure, storing a `Failure` value.
+  case failure(Failure)
+  
+  /// Evaluates the given transform closure when this `Result` instance is
+  /// `.success`, passing the value as a parameter.
+  ///
+  /// Use the `map` method with a closure that returns a non-`Result` value.
```

Comment:
Why can't `NewSuccess` be a value of type `Result` to give a `Result<Result, Failure>`?

I think what you're getting at is that the return value of the closure is used as the success value of a new `Result`, but that can be phrased more clearly (see below about the `- Returns:` documentation).

Reply:
It seems like `Result<Result, Failure>` would be confusing, or at least awkward to read, no? I've updated the rest of the inline docs following the suggestions you provided.